lib/winston/transports/file.js

```diff
@@ -678,13 +686,38 @@ module.exports = class File extends TransportStream {
 
     asyncSeries(tasks, () => {
       fs.rename(
-        path.join(this.dirname, `${basename}${ext}`),
+        path.join(this.dirname, `${basename}${ext}${isZipped}`),
         path.join(this.dirname, `${basename}1${ext}${isZipped}`),
         callback
       );
     });
   }
 
+  /**
+   * Compresses src to dest with gzip and unlinks src
+   * @param {string} src - path to source file.
+   * @param {string} dest - path to zipped destination file.
+   * @param {Function} callback - callback called after file has been compressed.
+   * @returns {undefined}
+   * @private
+   */
+  _compressFile(src, dest, callback) {
+    fs.exists(src, exists => {
+      if (!exists) return callback();
+
+      var gzip = zlib.createGzip();
+
+      var inp = fs.createReadStream(src);
+      var out = fs.createWriteStream(dest);
+
+      out.on('finish', () => {
```

Comment:
Can you test with the below code?
Because it can unlink the src file after 'close' event.
```suggestion
      inp.on('close', () => {
```

Reply:
Reason to wait for the `finish` event on the writable stream is to be sure the data has been fully written, before unlinking the source file. The `close` on the readable stream indicates, that data has been completely read,  but not that it has been completely processed.   
Therefore i'd keep it as it is.